compiler/damlc/daml-prim-src/GHC/Classes.daml

```diff
@@ -460,6 +463,27 @@ instance NumericScale 36 where numericScale _ = 36; numericScalePrivate _ = ()
 instance NumericScale 37 where numericScale _ = 37; numericScalePrivate _ = ()
 #endif
 
+#ifdef DAML_BIGNUMERIC
+
+-- | Types that can be represented by decimal literals in Daml.
+class IsNumeric t where
+    -- | Convert from `Numeric`. Raises an error if the number can't be
+    -- represented exactly in the target type.
+    fromNumeric : NumericScale m => Numeric m -> t
+
+    -- | Convert from `BigNumeric`. Raises an error if the number can't be
+    -- represented exactly in the target type.
+    fromBigNumeric : BigNumeric -> t
```

Comment:
We could keep it as an empty typeclass, or as a typeclass with a private method, but I think I prefer adding these methods. Exposing them here makes them easier to learn and more consistent (vs using the respective methods in DA.Numeric/DA.BigNumeric). And if we decide to let users define their own types with decimal literals, it would be easy to add them via this typeclass. This would also make Numeric literals more usable in some cases, since it can handle polymorphic scales via the typeclass.

Reply:
cool then let’s keep it